lib/xnvme_be_ramdisk_dev.c

```diff
@@ -0,0 +1,115 @@
+// re
+// Copyright (C) Mads Ynddal <m.ynddal@samsung.com>
+// SPDX-License-Identifier: Apache-2.0
+#ifndef _XOPEN_SOURCE
+#define _XOPEN_SOURCE 700
+#endif
+#include <xnvme_be.h>
+#include <xnvme_be_nosys.h>
+#ifdef XNVME_BE_RAMDISK_ENABLED
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <libxnvme_adm.h>
+#include <libxnvme_file.h>
+#include <libxnvme_spec_fs.h>
+#include <xnvme_be_ramdisk.h>
+#include <xnvme_dev.h>
+#include <xnvme_be_posix.h>
+
+void
+xnvme_be_ramdisk_dev_close(struct xnvme_dev *dev)
+{
+	struct xnvme_be_ramdisk_state *state;
+
+	if (!dev) {
+		return;
+	}
+
+	state = (void *)dev->be.state;
+	if (state->ramdisk) {
+		free(state->ramdisk);
+	}
+
+	memset(&dev->be, 0, sizeof(dev->be));
+}
+
+size_t
+_xnvme_be_ramdisk_dev_get_size(struct xnvme_dev *dev)
+{
+	struct xnvme_be_ramdisk_state *state = (void *)dev->be.state;
+	const struct xnvme_ident *ident = &dev->ident;
+
+	char *gb_marker = strrchr(ident->uri, 'G');
```

Comment:
See previous comments on moving up declaration and not doing logic which requires error-handling.